### DISCUSSION TITLE
myBuildSettings.ToolVersion

### DISCUSSION THREAD

**Discussion by crb3693** [cake-build/cake#2807] [Q&A] [answered]
I have a server that has .net framework 4.8 installed.  How do I use cake myBuildSettings.ToolVersion to specify to use .net framework 4.8?  It is not listed as an option on the page: https://cakebuild.net/api/Cake.Common.Tools.MSBuild/MSBuildToolVersion/

**Comment by Insire** [comment, marked as answer]
the msbuild tool isnt tied to the installed .net framework version. it either is installed as standalone tool or it is installed with visual studio.

you can find the standlone installer [here](https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&rel=16).

**Reply by crb3693** [reply]
If i install that, can i use the VS2019 option to build?

**Reply by Insire** [reply]
that should work, yes

**Reply by crb3693** [reply]
ok, so i installed the VS 2019 build tools and target VS2019 in cake but now i am still getting an error: 
error CS1525: Invalid expression term 'int' [c:\jenkins-sss\workspace\CPA Platform\Platform\CPA Platform\CPA Configuration Extractor\CPA Configuration Extractor.csproj]

I checked which language version of C# is default and its 8.0 so I am confused why "Inline variable declaration" is not compiling.  Also, other new language features like the => operator is working so that makes this even more confusing.

**Reply by Insire** [reply]
I'm not sure, but maybe you dont want to build this with msbuild, but perhaps with the dotnet tool.
